charmcraft/config.py

```diff
@@ -262,6 +262,35 @@ def validate_charm_type(cls, charm_type, values):
             raise ValueError("must be either 'charm' or 'bundle'")
         return charm_type
 
+    @classmethod
+    def expand_short_form_bases(cls, bases: List[Dict[str, Any]]) -> None:
+        """Expand short-form base configuration into long-form in-place."""
+        for index, base in enumerate(bases):
+            # Skip if already long-form. Account for common typos in case user
+            # intends to use long-form, but did so incorrectly (for better
+            # error message handling).
+            if (
+                "run-on" in base
+                or "run_on" in base
```

Comment:
Why do we check both `run-on` and `run_on`? Shouldn't the underscore be translated to dash on unmarshal?

Reply:
This check happens before the unmarshal, so I want to make sure we capture the mispelled case.